### PR TITLE
fix(ui): display correct invoice expire time

### LINKED ISSUE #2547: Invoice expire time displays incorrectly for invoices with distant expiry
Screenshot_124

It should say "expires" and not "expired", right?

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

Recheck invoice expire on a regular schedule to ensure that it works consistently for large expire times (setInterval has a max value which is less than the max possible expire time for an invoice).

## Motivation and Context:

fix #2547

## How Has This Been Tested?

Manually

## Types of changes:

Bug fix

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [x] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: renderer/components/Request/RequestSummary.js [hunk 1]
```diff
@@ -15,180 +16,164 @@ import { CryptoSelector, CryptoValue, FiatSelector, FiatValue } from 'containers
...
-    payReq: PropTypes.string.isRequired,
-    /** Show a notification. */
-    showNotification: PropTypes.func.isRequired,
-  }
+const RequestSummary = ({ invoice = {}, payReq, intl, showNotification, ...rest }) => {
+  const [isExpired, setIsExpired] = useState()
 
-  componentDidMount() {
-    const { payReq } = this.props
+  const scheduler = createScheduler()
```
Comment (korhaliv) [inline renderer/components/Request/RequestSummary.js]: should we use `ref` here? Probably we shouldn't re-create scheduler on every render

File: renderer/components/Request/RequestSummary.js [hunk 2]
```diff
@@ -15,180 +16,164 @@ import { CryptoSelector, CryptoValue, FiatSelector, FiatValue } from 'containers
...
+      })
     }
-
-    const { isExpired } = this.state
-    let decodedInvoice
-    try {
-      decodedInvoice = decodePayReq(payReq)
-    } catch (e) {
-      return null
+    // Clear schedular on unmount.
```
Comment (korhaliv) [inline renderer/components/Request/RequestSummary.js]: typo `schedular`

Comment (mrfelton) [reply renderer/components/Request/RequestSummary.js]: Updated.

File: renderer/components/Request/RequestSummary.js (hunk 1 above)
Comment (mrfelton) [reply renderer/components/Request/RequestSummary.js]: Yeah, totally. I'm surprised it was working as it was tbh. I've updated and rechecked and it looks to still be working correctly.

**Pull request merged**